## Questão 1

Necessitamos de um programa que, dado um raio R, informado pelo usuário, calcule:<br> 
(a) O comprimento da circunferência de raio R: C = 2 π R <br>
(b) A área do círculo de raio R: A = π R2 <br>
(c) O volume da esfera de raio R: V = ¾πR3

### Informações

Dados de entrada: raio do círculo <br>
Resultado esperado: obter informações sobre circunferência, area e volume <br>
O que o programa deve fazer: printar na tela as informações obtidas<br>

### Solução

- Entrada: obter o raio<br>
- Processamento: <br>- calcular a medida da circunferência<br>- calcular a área<br>- calcular o volume<br>
- Saída: exibir as informações obtidas no processamento<br>

In [ ]:
from math import pi


def get_circle(radius : float) -> dict:
    circunferency = 2 * pi * radius
    area = pi * radius**2
    volume = (3 * pi * radius**3) / 4

    circle_info = {
        'circunferency': circunferency,
        'area': area,
        'volume': volume
    }

    return circle_info


circle = get_circle(float(input('Digite o raio do círculo: ')))
print(
    f"Circunferência: {circle['circunferency']:.2f}\n"
    f"Area: {circle['area']:.2f}\n"
    f"Volume: {circle['volume']:.2f}"
)

### Casos de uso:

- Dados de entrada: raio = 5
- Resultado esperado: <br>

Circunferência: 31.42<br>
Area: 78.54<br>
Volume: 294.52<br>

- Dados de entrada: raio = 10
- Resultado esperado: <br>

Circunferência: 62.83<br>
Area: 314.16<br>
Volume: 2356.19<br> 


## Questão 2

Pedro tem um gato. O gato de Pedro só come ração. Pedro viaja com  muita frequência e, nessas ocasiões, pede a sua vizinha para alimentar seu gato.<br>Pedro gostaria de um programa que dada a quantidade de ração que o gato de Pedro  come por dia (em gramas) e a quantidade de dias que Pedro<br>irá passar viajando,  informe quantos sacos de ração ele precisa deixar com sua vizinha.  OBS: Um saco contém 1 (um) quilo de ração. 

### Informações

Dados de entrada: quantidade de ração(gramas/dia) que o gato come; quantidade de dias viajando<br>
Resultado esperado: quantos sacos de ração ele deve deixar com sua vizinha<br>
O que o programa deve fazer: printar na tela a quantidade de ração necessária<br>

### Solução

- Entrada: ração(gramas/dia); dias viajando<br>
- Processamento: obter total de ração necessária<br>
- Saída: exibir a informação obtida no processamento<br>

In [ ]:
racao_necessaria = lambda gramas_dia, dias_viajando : gramas_dia * dias_viajando

gramas = int(input('Qual a quantidade de gramas consumida por dia de ração? (apenas o número)\n: '))
dias = int(input('A viagem será de quantos dias? (apenas número)\n: '))

print(f'A quantidade de ração necessária será de {racao_necessaria(gramas, dias)} gramas')


### Casos de uso:

- Dados de entrada: gramas/dia = 100 gramas; dias viajando = 10 dias
- Resultado esperado: A quantidade de ração necessária será de 1000 gramas
<br>
- Dados de entrada: gramas/dia = 123 gramas; dias viajando = 45 dias
- Resultado esperado: A quantidade de ração necessária será de 5535 gramas

## Questão 3

Dizemos que um quadrado envolvente de um círculo é o menor quadrado  possível que contenha um círculo em seu interior.<br>Assumindo que esse quadrado está  alinhado com os eixos, implemente um programa que receba o canto superior  esquerdo<br> e o canto inferior direito do quadrado e calcule a coordenada do centro do  círculo, bem como seu raio

### Informações

Dados de entrada: vértice superior esquerdo; vértice inferior direito<br>
Resultado esperado: raio e coordenadas do centro<br>
O que o programa deve fazer: exibir o raio e as coordenadas do centro<br>

### Solução

- Entrada: p1, p2<br>
- Processamento: calcular raio e obter coordenadas do centro<br>
- Saída: exibir informações obtidas no processamento<br>

In [18]:
from math import sqrt


def get_circle_info(p1: list[float, float], p2: list[float, float]) -> dict:
    height = p1[1] - p2[1]
    width = p2[0] - p1[0]
    if height == width:
        center_position = [p1[0] - width/2, p1[1] - height/2]

        circle_info = {
            'radius': sqrt(width**2),
            'center_position': center_position
        }

        return f"Raio: {circle_info['radius']}\nCoordenadas centro: {circle_info['center_position']}"
    else:
        return f"Não é um quadrado"


print(get_circle_info(
    [int(input('x1: ')), int(input('y1: '))],
    [int(input('x2: ')), int(input('y2: '))]
))


Raio: 5.0
Coordenadas centro: [12.5, 7.5]


### Casos de uso:

- Dados de entrada: (5, 0); (0, 5)
- Resultado esperado:<br>
Raio: 5.0<br>
Coordenadas centro: [7.5, 2.5]
<br>

- Dados de entrada: (10, 5), (5, 10)
- Resultado esperado:<br>
Raio: 5.0<br>
Coordenadas centro: [12.5, 7.5]

### Questão 4

Implemente um programa para solicitar ao usuário dois valores inteiros,  um para a variável A e outro para a variável B. O programa deverá efetuar a troca dos  valores da variáveis de forma que a variável A passe a armazenar o valor que estava  anteriormente na variável B e a variável B passe a armazenar o valor que estava  anteriormente na variável A. Ao final, o programa deve exibir a seguinte frase: 
O novo valor de A é _______ e o novo valor de B é


### Informações

Dados de entrada: inteiro A; inteiro B<br>
Resultado esperado: A -> B; B -> A<br>
O que o programa deve fazer: Fazer uma troca entre os valores das variáveis<br>

### Solução

- Entrada: int A, int B<br>
- Processamento: A -> B; B -> A<br>
- Saída: Exibir a frase com o valor novo e antigo de A e B<br>

In [ ]:
def switcher(A: int, B: int) -> tuple[str]:
    values = [A, B]
    A = values[1]
    B = values[0]
    return (
        f'O novo valor de A é {A} e o novo valor de B é {B}\n'
        f'A antes -> {B}; A agora -> {A}\n'
        f'B antes -> {A}; B agora -> {B}'
    )

print(switcher(
    int(input('Digite o valor de A:')),
    int(input('Digite o valor de B:'))
))

### Casos de uso:

- Dados de entrada: A = 1; B = 2
- Resultado esperado:<br>
O novo valor de A é 2 e o novo valor de B é 1<br>
A antes -> 1; A agora -> 2<br>
B antes -> 2; B agora -> 1


- Dados de entrada: A = 12; B = 34
- Resultado esperado:<br>
O novo valor de A é 34 e o novo valor de B é 12<br>
A antes -> 12; A agora -> 34<br>
B antes -> 34; B agora -> 12